src/SFML/Window/Unix/MonitorImplX11.cpp

```diff
@@ -0,0 +1,193 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2024 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/Unix/Utils.hpp>
+#include <SFML/Window/Unix/MonitorImplX11.hpp>
+#include <SFML/Window/VideoMode.hpp>
+
+#include <SFML/System/Err.hpp>
+
+#include <X11/Xlib.h>
+#include <X11/extensions/Xrandr.h>
+
+#include <algorithm>
+#include <memory>
+#include <ostream>
+
+
+namespace sf::priv
+{
+////////////////////////////////////////////////////////////
+template <>
+struct XDeleter<XRRScreenConfiguration>
+{
+    void operator()(XRRScreenConfiguration* config) const
+    {
+        XRRFreeScreenConfigInfo(config);
+    }
+};
+
+
+////////////////////////////////////////////////////////////
+MonitorImplX11::MonitorImplX11(std::shared_ptr<Display>&& display, int screen) : m_display(std::move(display)), m_screen(screen)
+{
+}
+
+
+////////////////////////////////////////////////////////////
+std::unique_ptr<MonitorImpl> MonitorImplX11::getPrimaryMonitor() {
+    // Open a connection with the X server
+    auto display = openDisplay();
+    if (display)
+    {
+        // Retrieve the default screen number
+        return std::make_unique<MonitorImplX11>(std::move(display), DefaultScreen(display.get()));
+    }
+    else
+    {
+        // We couldn't connect to the X server
+        err() << "Failed to connect to the X server while trying to get the supported video modes" << std::endl;
+    }
+}
+
+
+////////////////////////////////////////////////////////////
+std::vector<VideoMode> MonitorImplX11::getFullscreenModes()
+{
+    std::vector<VideoMode> modes;
+
+    // Check if the XRandR extension is present
+    int version = 0;
+    if (XQueryExtension(m_display.get(), "RANDR", &version, &version, &version))
```

Comment:
Invert `if` and early return to reduce nesting

Reply:
I love this practice and hope one day we can enable the clang-tidy [`readability-else-after-return`](https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html) to programmatically enforce its usage. I've played with this before in SFML and it results in a _lot_ of code moving towards the left margin and reduces line count as well.

Reply:
Yes, for me it's a no-brainer. As soon as I see the early return, I know that I can focus on the rest of the function. 

With the nesting, my mind needs to keep track of the fact *"hey, an else is coming up eventually"* and it just makes everything harder to understand.

Fun fact: at least 2 professors in my uni who taught C were in the "never early return, only one return" camp. I feel sorry for the students that took that advice to heart without considering that the professor might be wrong.

Reply:
While I definitely agree and it is something I noticed as well, to keep this PR on-topic, I have moved the monitor X11 implementation onto its own branch: https://github.com/L0laapk3/SFML/tree/feature/monitor-unix where I will definitely be addressing your comments in the future.